apis/ec2/2016-11-15/waiters-2.json

```diff
@@ -506,6 +506,31 @@
         }
       ]
     },
+    "StoreImageTaskComplete": {
+      "delay": 5,
+      "operation": "DescribeStoreImageTasks",
+      "maxAttempts": 40,
+      "acceptors": [
+        {
+          "expected": "Completed",
+          "matcher": "pathAll",
+          "state": "success",
+          "argument": "StoreImageTaskResults[].StoreTaskState"
+        },
+        {
+          "expected": "Failed",
+          "matcher": "pathAny",
+          "state": "failure",
+          "argument": "store_image_task_results[].store_task_state"
```

Comment:
Are you sure this branch worked? I think this needs to be StoreImageTaskResults[].StoreTaskState

Reply:
you're definitely right. I built the resultant waiter by hand when i was testing (before i was aware of how the code generation worked), and converted it into the yaml format later. I clearly screwed that up on these last two.

Reply:
That change doesn't appear to alter the generated code, and it's identical to the hand-built one i've tested aside from using double instead of single quotes.

I haven't tested the failure state - i'm not sure how to force a failure of an image store task - but retry certainly works

Reply:
Depending on how quick it functions, you can possibly cancel it on another thread/terminal or from the console.

Reply:
I don't actualyl see an API to cancel it - only create and describe.

Reply:
I was able to test it by cancelling the multipart upload the task initiated. I received `block in poll': stopped waiting, encountered a failure state (Aws::Waiters::Errors::FailureStateError)` which seems like probably the correct behavior.